src/group_theory/free_abelian_group.lean

```diff
@@ -2,19 +2,68 @@
 Copyright (c) 2018 Kenny Lau. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Kenny Lau
-
-Free abelian groups as abelianization of free groups.
-
--- TODO: rewrite in terms of finsupp
 -/
+
 import algebra.group.pi
 import group_theory.free_group
 import group_theory.abelianization
+-- the below import is just for ℤ[empty] ≃+ unit
+import algebra.punit_instances
+-- the below import is just for ℤ[punit] ≃+ ℤ
+import algebra.module.basic
+/-!
+# Free abelian groups
+
+The free abelian group on a type `α`, defined as the abelianisation of
+the free group on `α`. The free abelian group on `α` is the left adjoint of the
+forgetful functor from abelian groups to types. A model for it is the functions
+`α → ℤ` which send all but finitely many `(a : α)` to `0`, under pointwise addition.
+
+## Main definitions
+
+Here we use the following variables: `(α β : Type*) (A : Type*) [add_comm_group A]`
+
+* `free_abelian_group α` : the free abelian group on a type `α`. As an abelian
+group it is `α →₀ ℤ`, the functions from `α` to `ℤ` such that all but finitely
+many elements get mapped to zero, however this is not how it is implemented.
+
+* `lift (f : α → A) : free_abelian_group α →+ A` : the group homomorphism induced
+by the map `f`.
+
+* `hom_equiv : (free_abelian_group α →+ A) ≃ (α → A)` : the bijection witnessing adjointness.
+
+* `map (f : α → β) : free_abelian_group α →+ free_abelian_group β` : functoriality
+    of `free_abelian_group`
+
+* `instance [monoid α] : semigroup (free_abelian_group α)`
+
+* `[comm_monoid α] : comm_ring (free_abelian_group α)`
+
+It has been suggested that we would be better off refactoring this file
+and using `finsupp` instead.
+
+## Main results
```

Comment:
Is this left empty on purpose? Could this section then not be removed?

Reply:
I kept expecting to find some results, but the main "results" are all definitions in some sense -- they are various constructions and equivs and they all contain data. I'll remove it!